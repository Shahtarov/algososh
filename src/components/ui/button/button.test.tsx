import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Button component", () => {
	it("button with text", () => {
		render(<Button text={"text"} />);

		const btn = screen.getByRole("button");

		expect(btn).toHaveTextContent(/text/i);
		expect(btn).toMatchSnapshot();
	});

	it("button without text", () => {
		render(<Button />);

		const btn = screen.getByRole("button");

		expect(btn).toHaveTextContent("");
		expect(btn).toMatchSnapshot();
	});

	it("locked button", () => {
		render(<Button disabled={true} />);

		const btn = screen.getByRole("button");

		expect(btn).toBeDisabled();
		expect(btn).toMatchSnapshot();
	});

	it("button with a loading indicator", () => {
		render(<Button isLoader={true} />);

		const btn = screen.getByRole("button");
		const img = screen.getByRole("img");

		expect(img).toBeInTheDocument();
		expect(btn).toMatchSnapshot();
	});

	it("button with callback", () => {
		const mockCallback = jest.fn();
		render(<Button onClick={mockCallback} />);

		const btn = screen.getByRole("button");
		fireEvent.click(btn);

		expect(mockCallback).toHaveBeenCalledTimes(1);
		expect(btn).toMatchSnapshot();
	});
});
