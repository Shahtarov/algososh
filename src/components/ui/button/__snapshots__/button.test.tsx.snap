// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component button with a loading indicator 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component button with callback 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component button with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button component button without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component locked button 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
